### PR TITLE
Filters out 'Complete' and 'No Access' work orders closed prior to appointment

### PR DESCRIPTION
**Pull request by teodor-hristoforov-hackney**
## Link to JIRA ticket

https://www.pivotaltracker.com/n/projects/2500129/stories/180277850

## Describe this PR

Filters out 'Complete' and 'No Access' work orders closed prior to appointment and add unit tests.

### CODE REVIEW

File: RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs [hunk 1]
```diff
@@ -81,7 +81,37 @@ public async Task<IEnumerable<OperativeWorkOrderListItem>> Execute(OperativeWork
                 _logger.LogInformation($"{workOrders.Count()} work orders for operative {opSearch.PayrollNumber}: [${string.Join(", ", workOrders.Select(w => w.Id.ToString()))}]");
             }
 
+            DateTime earliestAppointementDate = GetEarliestAppointmentFromWorkOrders(workOrders);
+
+            workOrders = workOrders.Where(wo =>
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs:86]: I think there is a logical error here. The requirement is:
"Filter out 'Complete' and 'No Access' work orders that closed prior to appointment"
whereas you have designed:
"Filter out 'Complete' and 'No Access' work orders, or those that closed prior to appointment"

Comment (teodor-hristoforov-hackney) [reply RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs:86]: I fixed the issue and updated the unit tests to reflect that. Let me know if you spot anything else out of order

File: RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs [hunk 2]
```diff
@@ -81,7 +81,33 @@ public async Task<IEnumerable<OperativeWorkOrderListItem>> Execute(OperativeWork
                 _logger.LogInformation($"{workOrders.Count()} work orders for operative {opSearch.PayrollNumber}: [${string.Join(", ", workOrders.Select(w => w.Id.ToString()))}]");
             }
 
+            DateTime earliestAppointementDate = GetEarliestAppointmentFromWorkOrders(workOrders);
+
+            var excludedWorkOrders = workOrders.Where(wo =>
+            (wo.StatusCode == WorkStatusCode.Complete && wo.ClosedDate > earliestAppointementDate ||
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs:87]: prior to I think is < and not >

Comment (teodor-hristoforov-hackney) [reply RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs:87]: This one grabs the excluded work orders for easier readability and then does ` workOrders = workOrders.Except(excludedWorkOrders);`

Comment (Asura5) [reply RepairsApi/V2/UseCase/ListOperativesWorkOrdersUseCase.cs:87]: Yes and you want to filter out ones that are prior to......

**Review by Asura5 — APPROVED** [review]

**Pull request merged**